setup.sh

```diff
@@ -0,0 +1,283 @@
+
+        msg "installed postgress, should now be on path"
+        # TODO replace this with cli open.
+        open -a postgres
+        sleep 5
+        
+        if [ $? -ne 0 ]; then
+            error "Could not install postgres."
+        fi
+        verbose "installed postgres"
+        else 
+            sudo apt-get install -y postgresql \
+                            postgresql-contrib \
+                            libpq-dev
+
+            echo "Ensure pg_config is on the PATH, and then login to Postgres. You will need to locate where `apt-get` has installed your Postgres binary and add this to your PATH. You can use: whereis psql for that, but ensure you add the directory and not the executable to the path"
+            read -p "Press enter to continue"
+
+            export PATH=/usr/bin:$PATH
+            sudo -u postgres createuser --superuser $USER
+            sudo -u postgres createdb $USER
+        fi
+
+        psql -c "CREATE ROLE itig WITH CREATEDB PASSWORD 'd872\$dh' LOGIN;"
+        if [ $? -ne 0 ]; then
+            error "Could not install postgres. Please ensure psql is on the path, and then rerun this script."
+            exit 1
+        fi
+
+        verbose "installed postgres"
+}
+
+# 
+# Install native tools
+# 
+install_native_tools () {
+    msg "Installing native tools"
+    if isMac; then
+        brew install imagemagick libmagic ghostscript
+        msg "Trying to install pygments with easy_install, please enter your password"
+        sudo easy_install Pygments
+    else
+        sudo apt-get install -y ghostscript \
+                       imagemagick \
+                       libmagickwand-dev \
+                       libmagic-dev \
+                       python-pygments
+    fi
+    if [ $? -ne 0 ]; then
+        error "Could not install native tools, please review the terminal window for details."
+        error "Packages may have already been installed, attempting to continue with setup."
+    else verbose "installed native tools"
+    fi
+}
+
+# 
+# Install doubtfire gem dependencies
+# 
+install_dfire_dependencies () {
+    msg "Installing doubtfire dependencies"
+    gem install bundler
+    bundler install --without production replica
+    rbenv rehash
+    source ~/.bashrc
+
+    msg "Populating database"
+    rake db:create
+    rake db:populate
+
+    if [ $? -ne 0 ]; then
+        error "Could not populate database."
+        exit 1
+    fi
+
+    verbose "populated database"
+}
+
+# 
+# Install doubtfire gem dependencies
+# 
+install_dstil_overcommit () {
+    curl -s https://raw.githubusercontent.com/dstil/dotfiles/master/bootstrap | bash
+
+    gem install overcommit
+    rbenv rehash
+    overcommit --install
+    dstil --sign
+
+    verbose "installed dstil hooks."
+}
+
+install_latex () {
+    if isMac; then
+        read -r -p "LaTeX is required for PDF generation, it could take up to several hours to install, would you like to install LaTeX now? [y/N] " response
```

Comment:
But apt-get doesn't specify that LaTeX install is needed for PDF generation? Maybe echo that first and let `apt-get -y` handle it, whereas for mac a simple y/n will do.